iroh-api/src/api.rs

```diff
@@ -48,9 +48,21 @@ pub trait Api {
         ipfs_path: &IpfsPath,
     ) -> LocalBoxStream<'_, Result<(RelativePathBuf, OutType)>>;
 
-    fn add_file<'a>(&'a self, path: &'a Path, wrap: bool) -> LocalBoxFuture<'_, Result<Cid>>;
-    fn add_dir<'a>(&'a self, path: &'a Path, wrap: bool) -> LocalBoxFuture<'_, Result<Cid>>;
-    fn add_symlink<'a>(&'a self, path: &'a Path, wrap: bool) -> LocalBoxFuture<'_, Result<Cid>>;
+    fn add_file(
+        &self,
+        path: &Path,
+        wrap: bool,
+    ) -> LocalBoxFuture<'_, Result<LocalBoxStream<'static, Result<AddEvent>>>>;
```

Comment:
Let's leave it for now, but why do these have to be the Local variants anyway? As an user of the API that would be inconvenient.

Will take a look.